internal/util/url.go

```diff
@@ -75,11 +76,10 @@ func (f ReadUnmounter) Close() error {
 }
 
 // FetchResourceAsReader returns a ReadCloser to the data at the url specified. The caller is responsible for
-// closing the reader.
-func FetchResourceAsReader(l *log.Logger, u url.URL) (io.ReadCloser, error) {
+// closing the reader. client is only necessary if the url is http(s).
+func FetchResourceAsReader(l *log.Logger, client HttpClient, u url.URL) (io.ReadCloser, error) {
```

Comment:
I know this has already been lgtm'd by crawford, but I'm a bit bothered by the passing around of the `HttpClient` everywhere, then we arrive here to learn that the `HttpClient` may not even be applicable for non-http urls.

So I'm proposing that we not have this `HttpClient` dispersal, and instead maybe create a new url type for the fetchers which groups timeout values with the url.URL.  Then these functions which perform the gets on the url can ask the new url type for the timeouts when applicable.  If no timeouts are set on the url, use the appropriate defaults.  Or maybe when the new url type is created it can have the defaults set... that's just an implementation detail though, whatever seems more natural makes sense.

The url is already passed where appropriate, we just need to get the timeouts from the config into the url where we begin the retrievals.

Hope that makes sense!  Correct me if I'm being silly, I haven't looked at golang for a few months now.